examples/MIFARE/mifareReader.device.nut

```diff
@@ -0,0 +1,85 @@
+#require "PN532.class.nut:1.0.0"
+#require "PN532MifareClassic.class.nut:1.0.0"
+
+static USER_ID_ADDRESS = 0x02;
+
+// Label all pins
+local spi = hardware.spi257;
+local ncs = hardware.pin1;
+local rstpd = hardware.pin3;
+local irq = hardware.pin4;
+
+// Pre-configure SPI bus
+spi.configure(LSB_FIRST | CLOCK_IDLE_HIGH, 2000);
+
+// Returns whether this is a MIFARE Classic 1k tag
+// A base list of identifiers can be found at http://nfc-tools.org/index.php?title=ISO14443A
+function isMifareClassic1k(tagData) {
+    return tagData.SENS_RES[0] == 0x00 && tagData.SENS_RES[1] == 0x04 && tagData.SEL_RES == 0x08;
+}
+
+function pollCallback(error, numTagsFound, tagData) {
+    if(error != null) {
+        server.error(error);
+        return;
+    }
+        
+    if(numTagsFound > 0 && isMifareClassic1k(tagData)) {
+        mifareReader.authenticate(tagData.NFCID, 0x2, PN532MifareClassic.AUTH_TYPE_A, null, function(error, authenticated) {
+            if(error != null) {
+                server.error(error);
+                return;
+            }
+            
+            if(authenticated) {
+                mifareReader.read(0x2, function(error, data) {
+                    if(error != null) {
+                        server.error(error);
+                        return;
+                    }
+                    
+                    server.log("Found tag with user ID: ")
+                    server.log(data);
+                    
+                    // Repeat for next tag in a second
+                    imp.wakeup(1, function() {
```

Comment:
Nothing other than waiting a second to poll again... anything smarter seemed a bit application-specific.
It wouldn't be too hard to make up an application and implement this, however.